config/config.go

```diff
@@ -292,6 +292,75 @@ func GetAllPrefixes() []string {
 	return prefixes
 }
 
+// This function is for discovering federations as specified by a url during a pelican:// transfer.
+// this does not populate global fields and is more temporary per url
+func DisoverUrlFederation(federationDiscoveryUrl string) (FederationDiscovery, error) {
```

Comment:
Too much duplication with DidcoverFederation.

Have DiscoverFederation use this function when it is ready.

Needs a unit test.